lib/containers/utils.pm

```diff
@@ -179,4 +179,13 @@ sub test_built_img {
     assert_script_run("rm -rf /root/templates");
 }
 
+sub is_sle_container_supported {
```

Comment:
I read this function and I don't really understand what it does.

You can run SLE containers just fine everywhere, mostly independent of host OS and environment as that's what they're for.

Is this about registration and SUSEConnect? If so, this should be mentioned, maybe `are_sle_repos_available` or so.

In its current state this means that jobs cloned from o3 or osd behave significantly different. Is it possible to do feature detection somehow?

Reply:
there is an answer somewhere else in the conversations about that. but yes. Containers can run everywhere. But you can build. So yes it is about registration and SUSEConnect. in general even if it is possible as you had pointed (see https://github.com/SUSE/container-suseconnect#building-images-on-non-sle-distributions) is not supported official. And thus the name. 

What do you mean "feature detection"? i think there are some lines that they try to do this(if i understand) but it did not help in this case. short story, i think this is a very simple solution to complain with the current requirements. i dont know if it helps but i think the initial problem showed up when we decide to run Centos, Ubuntu, opensuse in OSD. 
What it does 

```
false if you try to test sle base container on o3 which run opensuse distros
false if you try to test sle base container on osd but host is not sle
true otherwise
```

Reply:
> And thus the name.

In that case I recommend `can_build_sle_container` or so.

> What do you mean "feature detection"?

Not deducing the result based on version checks or similar, but actually checking directly for the needed capability.
In this case, a check whether the system is registered would work, so something like `check_var('SCC_REGCODE')`, which would show whether the system is registered.

Reply:
> Not deducing the result based on version checks or similar, but actually checking directly for the needed capability.
> In this case, a check whether the system is registered would work, so something like `check_var('SCC_REGCODE')`, which would show whether the system is registered.

@Vogtinator unfortunately you **can not** judge about such thing based just on variable - checkout verification run for this PR : http://aquarius.suse.cz/tests/3882#settings
1. DISTRI=sle so `is_sle` will return **True**
2. SCC_REGCODE variable is presented 
3. `HDD_1=opensuse-15.2-x86_64-695.1-textmode@64bit.qcow2` so actually it is openSUSE ...

Reply:
Where is the `openqa.opensuse.org` part @b10n1k? It's missing in the current version of code.

Reply:
@pdostal i have removed it. i do not think is needed it. But i have not finished yet. so give me some time to run some VRs and i will ping you at the chat

Reply:
Then it seems like `return check_os_release('sle', 'PRETTY_NAME')` would be correct?

Reply:
agreed with @pdostal to provide a followup PR with enhanced `check_os_release` as he has a very good proposal.